crates/rolldown_common/src/inner_bundler_options/mod.rs

```diff
@@ -105,11 +106,24 @@ fn deserialize_treeshake<'de, D>(deserializer: D) -> Result<TreeshakeOptions, D:
 where
   D: Deserializer<'de>,
 {
-  let deserialized = Option::<bool>::deserialize(deserializer)?;
-  match deserialized {
-    Some(false) => Ok(TreeshakeOptions::False),
-    Some(true) | None => Ok(TreeshakeOptions::Option(types::treeshake::InnerOptions {
-      module_side_effects: types::treeshake::ModuleSideEffects::Boolean(true),
-    })),
+  let value = Option::<Value>::deserialize(deserializer)?;
```

Comment:
Not for this PR. We could do this https://github.com/rolldown/rolldown/pull/1681#discussion_r1686188850 later.

Reply:
We have used the generated shchema to validate options:

https://github.com/rolldown/rolldown/blob/2d2974e7b185b63cd7593093114e815e77b0d2a3/crates/rolldown_testing/src/test_config.rs#L39